.github/workflows/release_otelbench.yaml

```diff
@@ -0,0 +1,106 @@
+run-name: release-otelbench
+name: Release otelbench docker image on elastic container library
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Version of the otelbench image to release, should follow format "vN.N.N".
+        required: true
+  push:
+    branches:
+      - main
+    paths:
+      - ./loadgen/cmd/otelbench/Makefile
+
+jobs:
+  check-release:
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.get-new-version.outputs.new_version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: '0'
+
+      - name: Checks if release is required.
+        id: get-new-version
+        run: ./scripts/check_release.sh "${{ github.event_name }}" "${{ github.event.inputs.version }}"
+
+  release-otelbench:
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    needs:
+      - check-release
+    if: needs.check-release.result == 'success'
+    permissions:
+      contents: write
+    env:
+      ELASTIC_REGISTRY: "docker.elastic.co/observability-ci"
+      IMAGE_NAME: "otelbench"
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: '0'
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+          cache: 'false'
+
+      - name: Cache go tools
+        id: go-tools-cache
+        uses: actions/cache@v4
+        with:
+          path: ./.tools
+          key: go-tools-${{ runner.os }}-${{ hashFiles('internal/tools/go.sum') }}
+
+      # If there is no cache with the go tools,
+      # install them. We need the changelog
+      # tool for this workflow to work.
+      - name: Install tools
+        if: steps.go-tools-cache.outputs.cache-hit != 'true'
+        run: |
+          make install-tools
+
+      - name: Update otelbench changelog and delete changelog fragments
+        working-directory: ./loadgen/cmd/otelbench
+        run: ../../../.tools/chloggen update --version ${{ needs.check-release.outputs.version }}
+
+      - name: Update Makefile version if dispatched
+        if: github.event_name == 'workflow_dispatch'
+        working-directory: ./loadgen/cmd/otelbench
+        run: sed -i "s/^VERSION := .*/VERSION := ${{ github.event.inputs.version }}/" Makefile
+
+      - name: Commit changes
+        run: |
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
```

Comment:
You won't be able to push changes, but you can reuse one of our reusable gh actions to help with the config settings.

See https://github.com/elastic/oblt-actions/tree/main/git/setup

Regarding pushing changes, a few questions:

1) If this project uses GH actions for the CI, then it should not use the GitHub actions GH Secret but a GH app.
2) Otherwise, GH workflows won't run for events created by the GH actions bot - to avoid recurrence.
3) I cannot see the section `permissions` with the required scopes `contents: write`

Reply:
> I cannot see the section permissions with the required scopes contents: write

It's in the job itself:

https://github.com/elastic/opentelemetry-collector-components/blob/620724e0c765078fa59b8bddef4550826f2bdee7/.github/workflows/release_otelbench.yaml#L37-L38

Reply:
> If this project uses GH actions for the CI, then it should not use the GitHub actions GH Secret but a GH app.
Otherwise, GH workflows won't run for events created by the GH actions bot - to avoid recurrence.

I am a bit confused over this. This part of the workflow needs to delete files in the repository, and update a file. I understand now that it needs to create a PR, right? 

After the changelog is updated, then it creates a docker image and it pushes the image to elastic private registry.

Reply:
Aha, cool, thanks for this. Do you mind adding the section `contents: read` at the top?

```yaml

permissions:
  contents: read

jobs:
```

We use the least-permissive access by default.

Reply:
I have changed it to this:

https://github.com/elastic/opentelemetry-collector-components/blob/b579ae97bb8087e8aa0c6763e4f5e6c8745d7044/.github/workflows/release_otelbench.yaml#L82-L115

I don't know what should be inside `github-token: ${{ secrets.GITHUB_TOKEN }} `.

I am open to follow a different approach too, if you think it is best.

Reply:
I have sent you an invite to sync @v1v . Feel free to adjust the time if it does not fit you.

Reply:
I have updated the PR now @v1v so that:

- It fetches the github token
- Uses a script to create the PR, similarly to your reference

Reply:
@v1v Following our discussion, I have pushed a new commit:

- Uses the commit instead of the tag
- Splits the jobs between two workflows:
  - One to open a PR to update the version/changelog
  - Another one that gets triggered once the CHANGELOG.md is updated, and publishes a new docker image

Does it look good now?